fedimint-build/src/lib.rs

```diff
@@ -1,21 +1,31 @@
 use std::env;
-use std::fs::File;
+use std::io::{self, Write};
 use std::process::Command;
 
+const FAKE_COMMIT_HASH: &str = "01234569afbe457afa1d2683a099c7af48a523c1";
+
 pub fn set_code_version() {
     if env::var_os("CODE_VERSION").is_none() {
-        let git_hash = if File::open("./../../.git/HEAD").is_ok() {
-            let output = Command::new("git")
-                .args(["rev-parse", "HEAD"])
-                .output()
-                .unwrap();
-            String::from_utf8(output.stdout).unwrap()
-        } else {
-            // we set a fake hash here and make it easily recognizable
-            // by giving it a prefix (0x0123456)
-            String::from("01234569afbe457afa1d2683a099c7af48a523c1")
-        };
-        println!("cargo:rustc-env=CODE_VERSION={git_hash}");
+        let output = Command::new("git").args(["rev-parse", "HEAD"]).output();
+
+        match output {
+            Ok(output) => {
+                if output.status.success() {
+                    let git_hash = String::from_utf8_lossy(&output.stdout).trim().to_string();
+                    println!("cargo:rustc-env=CODE_VERSION={}", git_hash);
+                } else {
+                    let error_message = String::from_utf8_lossy(&output.stderr);
+                    io::stderr().write_all(error_message.as_bytes()).unwrap();
```

Comment:
I would use Command::spawn with stderr set to inherit and stdout set to pipe

Reply:
I'm not sure what value command::spawn adds here since you immediately need to wait for the child process to resolve. I was unable to find a construction I liked better than command::new. Happy to revisit if you disagree.